### PR TITLE
[RW-6272][risk=low][Spring 2 Upgrade Step1 ]: Upgrade hibernate version

### PR DESCRIPTION
**Pull request by yonghaoy**
Description:
**This upgrade will cause JPA return null if we forgot to set alias match to DTO'a variable name.** And it is the part I am worried the most for the entire Spring upgrade. 
What happened here is: If using projection with custom query, sometimes JPA does not know how to map the query column to interface DTO, so we have to explicated use alias.  
What I did here is:
Search all Dao classes for `interface` class, then add alias for them. That is actually better than I expected, as there are not too many closed projections. I also added a small test for adminUserTable.
Seems all tests are passing(15 failures without this). 
This seems to be fine for now and will unblock Spring upgrade. But I am not sure moving forward how we guard/test/prevent such things. But those questions seems out of spring upgrade scope. 

Few things I want to check:

1. Is this the right risk level? 
2. Is what I did sufficient, how confident we are about this PR change?
3. When to merge this PR? Still wait till next release? I am 80% confident now because I think I found all closed projections. 
4. Besides this, I think the reset Spring 2 upgrade are all low risk... 

(also being discussed in slack channel)

Thanks! 

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/db/dao/UserDao.java
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/db/dao/UserDao.java:151]: Thanks for adding this!  Could you add a little more about why, to guide future maintainers?  Specifically the fact that this will cause null results if incorrect.

**Review by jmthibault79 — COMMENTED** [review]
looks good so far but I think I won't be confident about this change until we test all fields for all nativeQueries with projections.  (I am not familiar with the distinction between "closed projection" and other kinds of projection so I can't comment about that)

I don't mean that we necessarily need new tests for this.  I would expect that many of these fields are already adequately covered by existing tests.  But we should confirm this.

Going forward we should establish some guidelines to protect against this null problem (maybe it's simply "don't use nativeQuery with projection"!) but I don't consider that to be blocking.

**Comment by yonghaoy** [comment]
> looks good so far but I think I won't be confident about this change until we test all fields for all nativeQueries with projections. (I am not familiar with the distinction between "closed projection" and other kinds of projection so I can't comment about that)
> 
> I don't mean that we necessarily need new tests for this. I would expect that many of these fields are already adequately covered by existing tests. But we should confirm this.
> 
> Going forward we should establish some guidelines to protect against this null problem (maybe it's simply "don't use nativeQuery with projection"!) but I don't consider that to be blocking.

I am new to Spring Data world, still trying to figure out those terminologies. Now I think I have a better understanding now. 

1. Native query works fine, not broken. 
2. Close Projection is the wrong word, sorry for the confusion.
3. What is the real problem is Interface-Based Projections with @query. 
Like mentioned here: https://stackoverflow.com/questions/40194614/spring-data-jpa-projection-selected-fields-from-the-db
Search around, seems the general advice for this is just always use alias!

**Comment by jmthibault79** [comment]
oh no, so it's not just nativeQuery?  Now I am more concerned.

From the  SO link:
> when I changed projections from interfaces to pojo classes it generated SQL properly.

That seems worth investigating, too.

**Comment by yonghaoy** [comment]
> oh no, so it's not just nativeQuery? Now I am more concerned.
> 
> From the SO link:
> 
> > when I changed projections from interfaces to pojo classes it generated SQL properly.
> 
> That seems worth investigating, too.

Actually I feel less concerned because we did not use that patten (interface projection + @query), compare with nativequery. 
What are the things you are concerned, and what suggestion you have that we should do pre/after merging to prevent & test JPA changes like this?

**Comment by yonghaoy** [comment]
> oh no, so it's not just nativeQuery? Now I am more concerned.
> 
> From the SO link:
> 
> > when I changed projections from interfaces to pojo classes it generated SQL properly.
> 
> That seems worth investigating, too.

For pojo, I think they refer to our this pattern: https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/db/model/DbUser.java#L99
Which also works fine, and I think that because we annotate with @column hence Spring knows how to do the mapping. 
I did not find a good way to improve interface pattern....

**Comment by jmthibault79** [comment]
We do use `@Query` + projection for JPQL queries (non-nativeQuery).  For an example, see `BillingProjectBufferEntryDao.getBillingBufferGaugeData()`.  So I would like to add all of those to the tests we're inspecting for this process as well.

DbUser is an `@Entity` and a `@Table` so it is not a POJO.  My understanding of what it was trying to say is: try using a `class` instead of an `interface` when using projections.  However my initial experiment with doing that did not succeed.  Here are some links I found which might be helpful - but I have not looked at them yet.
* https://www.baeldung.com/spring-data-jpa-projections
* https://thorben-janssen.com/spring-data-jpa-query-projections/
* https://thorben-janssen.com/projections-with-jpa-and-hibernate/

File: api/src/test/java/org/pmiops/workbench/db/dao/UserDaoTest.java
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/db/dao/UserDaoTest.java:75]: nice

**Review by jmthibault79 — APPROVED** [review]
Thanks for the new tests!

I looked for every projection I could find and I counted only 6 of them - so that's good to know we don't have many.

Please add a test for WorkspaceDao.getWorkspaceCountGaugeData() - looks like the fields of WorkspaceCountByActiveStatusAndTier might not be fully tested yet.

Thanks for doing this!

**Pull request merged**
